### PR TITLE
Feature/removing url parameter max length

### PR DESCRIPTION
**Pull request by SNiukalov**
Scripts to check  [#2702](https://github.com/smartdevicelink/sdl_core/issues/2702)

This PR is **ready** for review.

### Summary
Initial scripts for remove url max length

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0200-Removing-URL-Param-Max-Length.md
+--
+-- Description: Check processing of OnSystemRequest notification with different length of url
+--
+-- In case:
+-- 1. HMI sends OnSystemRequest notification with out of lower bound value in url parameter
```
Comment (GetmanetsIrina) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua]: `out of lower bound value in url parameter` -> length of url is less than min value

File: test_scripts/API/RemoveUrlParameterMaxLength/002_StartStream.lua [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+--
+-- Description: Check processing of Navigation.StartStream request with url length more than 500 characters
+--
+-- In case:
+-- 1. VideoStreamConsumer is set to file value
+-- 2. VideoStreamFile is set to value with length in 255 characters
+-- 3. AppStorageFolder is set to value with length in 300 characters
+-- 4. Video service starts
+-- SDL does:
+-- - sends Navigation.StartStream(<pathToSDL/AppStorageFolder + file name>)
```
Comment (GetmanetsIrina) [inline test_scripts/API/RemoveUrlParameterMaxLength/002_StartStream.lua]:
```suggestion
-- - sends Navigation.StartStream(<pathToSDL/AppStorageFolder>/<file name>)
```

File: test_scripts/API/RemoveUrlParameterMaxLength/003_StartAudioStream.lua [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+--
+-- Description: Check processing of Navigation.AudioStreamConsumer request with url length more than 500 characters
+--
+-- In case:
+-- 1. AudioStreamConsumer is set to file value
+-- 2. AudioStreamFile is set to value with length in 255 characters
+-- 3. AppStorageFolder is set to value with length in 300 characters
+-- 4. Audio service starts
+-- SDL does:
+-- - sends Navigation.StartAudioStream(<pathToSDL/AppStorageFolder + file name>)
```
Comment (GetmanetsIrina) [inline test_scripts/API/RemoveUrlParameterMaxLength/003_StartAudioStream.lua]:
```suggestion
-- - sends Navigation.StartAudioStream(<pathToSDL/AppStorageFolder>/<file name>)
```

**Comment by GetmanetsIrina** [comment]
@yang1070, all comments are processed. Please review.

**Comment by SNiukalov** [comment]
@yang1070 please approve these changes

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0200-Removing-URL-Param-Max-Length.md
+--
+-- Description: Check processing of OnSystemRequest notification with different length of url
+--
+-- In case:
+-- 1. HMI sends OnSystemRequest notification with length of url is less than min value
+-- SDL does:
+-- - ignore notification and not send it to mobile app
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:9]:
```suggestion
-- - ignore the  notification and does not send it to mobile app
```

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0200-Removing-URL-Param-Max-Length.md
+--
+-- Description: Check processing of OnSystemRequest notification with different length of url
+--
+-- In case:
+-- 1. HMI sends OnSystemRequest notification with length of url is less than min value
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:7]:
```suggestion
-- 1. HMI sends OnSystemRequest notification with a URL shorter than the minimum required length.
```

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+-- SDL does:
+-- - ignore notification and not send it to mobile app
+-- 2. HMI sends OnSystemRequest notification with in bound value in url parameter
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:10]:
```suggestion
-- 2. HMI sends OnSystemRequest notification with a URL longer than/the same length as the minimum required length.
```

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/RemoveUrlParameterMaxLength/commonRemoveUrlParameterMaxLength')
+
+--[[ Local Variables ]]
+local expected = 1
+local notExpected = 0
+local outOfMinLength = ""
+local minlength = "u"
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:22]: I think `minlength` implies an integer value. However, this value is used as a sample string for the test case
```suggestion
local minLengthString = "u"
```

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+-- - send notification with received url to mobile app
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/RemoveUrlParameterMaxLength/commonRemoveUrlParameterMaxLength')
+
+--[[ Local Variables ]]
+local expected = 1
+local notExpected = 0
+local outOfMinLength = ""
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:21]:
```suggestion
local lessThanMinLengthString = ""
```

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("RAI", common.registerApp)
+common.Step("Activate App", common.activateApp)
+
+common.Title("Test")
+
+common.Step("OnSystemRequest url outOfMinLength", OnSystemRequest, { outOfMinLength, notExpected })
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:42]:
```suggestion
common.Step("OnSystemRequest url lessThanMinLengthString", OnSystemRequest, { lessThanMinLengthString, notExpected })
```

File: test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+common.Step("OnSystemRequest url minlength", OnSystemRequest, { minlength, expected })
```
Comment (ShobhitAd) [inline test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua:43]:
```suggestion
common.Step("OnSystemRequest url minLengthString", OnSystemRequest, { minLengthString, expected })
```

File: test_sets/remove_url_paramter_max_lenght.txt [hunk 11]
```diff
@@ -0,0 +1,3 @@
+./test_scripts/API/RemoveUrlParameterMaxLength/001_OnSystemRequest.lua
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_sets/remove_url_paramter_max_lenght.txt]: @ShobhitAd Done in 1c68830

**Comment by dboltovskyi** [comment]
@ShobhitAd Please notice all comments have been processed.

**Pull request merged**